sql/core.go

```diff
@@ -399,6 +399,38 @@ type ProjectedTable interface {
 	Projections() []string
 }
 
+// HistogramBucket represents a bucket in a histogram
+// inspiration pulled from MySQL and Cockroach DB
+type HistogramBucket struct {
+	LowerBound float64 // inclusive
+	UpperBound float64 // inclusive
+	Frequency  float64
+}
+
+// Histogram is all statistics we care about for each column
+type Histogram struct {
+	Buckets       []*HistogramBucket
+	Mean          float64
+	Min           float64
+	Max           float64
+	Count         uint64
+	NullCount     uint64
+	DistinctCount uint64
+}
+
+// HistogramMap is a map from column name to associated histogram
+type HistogramMap map[string]*Histogram
+
+// TableStatistics will hold all the statistics on a table we care about
```

Comment:
TableStatistics provides access to statistical information about the values stored in a table